src/test/java/com/amihaiemil/camel/ScalarTest.java

```diff
@@ -104,6 +104,9 @@ public void equalsAndHashCode() {
 
         MatcherAssert.assertThat(firstScalar, Matchers.equalTo(secondScalar));
         MatcherAssert.assertThat(secondScalar, Matchers.equalTo(firstScalar));
+        MatcherAssert.assertThat(firstScalar, Matchers.equalTo(firstScalar));
+        MatcherAssert.assertThat(firstScalar, Matchers.equalTo(null));
```

Comment:
@salikjan This has to be a negation. So firstScalare !equals null. I think you can put ``Matchers.equalTo(null)`` inside ``Matchers.not``, or something like that,